src/main/scala/mesosphere/marathon/core/matcher/base/util/OfferOperationFactory.scala

```diff
@@ -47,78 +47,99 @@ class OfferOperationFactory(
       .build()
   }
 
-  def reserve(reservationLabels: ReservationLabels, resources: Seq[Mesos.Resource]): Mesos.Offer.Operation = {
-    val reservedResources = resources.map { resource =>
-
-      val reservation = ReservationInfo.newBuilder()
-        .setLabels(reservationLabels.mesosLabels)
-        .setPrincipal(principal)
-
-      Mesos.Resource.newBuilder(resource)
-        .setRole(role)
-        .setReservation(reservation)
-        .build()
+  private def getProviderId(resource: Mesos.Resource): Option[String] = {
+    if (resource.hasProviderId && resource.getProviderId.hasValue) {
+      Some(resource.getProviderId.getValue)
+    } else {
+      None
```

Comment:
this pattern seems to be repeated...  should we have a common function?

Reply:
Yes, I mentioned it in the slack channel. I will do it, but I just don't have time for it right, in hope that we land it today.

Reply:
I won't forget about it :)